### PR TITLE
Fixing some miscellaneous TODOs

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

There are several `TODO`s in the ARMI codebase. I found some that were low-hanging fruit to fix or test/delete if they were defunct.

## Why is the change being made?

Our docs say we shouldn't keep `TODO`s in the codebase. Also, some of these were questions that needed testing, so I just did the testing.

There are more left, but perfect is the enemy of better.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/bookkeeping/db/layout.py [hunk 1]
```diff
@@ -851,18 +851,11 @@ def replaceNonsenseWithNones(data: numpy.ndarray, paramName: str) -> numpy.ndarr
             if isNone[i].all():
                 result[i] = None
             elif isNone[i].any():
-                # TODO: This is not symmetric with the replaceNonesWithNonsense impl.
-                # That one assumes that Nones apply only at the highest dimension, and
-                # that the lower dimensions will be filled with the magic None value.
-                # Non-none entries below the top level fail to coerce to a serializable
-                # numpy array and would raise an exception when trying to write. TL;DR:
-                # this is a dead branch until the replaceNonesWithNonsense impl is more
-                # sophisticated.
```
Comment (albeanth) [inline armi/bookkeeping/db/layout.py:860]: I guess `replaceNonesWithNonsense` is now sufficiently sophisticated? 🤷‍♂️

**Review by albeanth — COMMENTED** [review]
Just 1 comment on settings not getting moved to physics. There's lots of physics specific settings listed in globalSettings.

File: armi/settings/fwSettings/globalSettings.py [hunk 2]
```diff
@@ -17,9 +17,6 @@
 
 This should contain Settings definitions for general-purpose "framework" settings. These
 should only include settings that are not related to any particular physics or plugins.
-
-TODO: There are lots of settings in here that violate the above rule, which still need
-to be migrated to their respective plugins: they are clearly separated for review.
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/settings/fwSettings/globalSettings.py:22]: If you look down the file a few dozen lines, we have another TODO that explicitly points to the parameters:

https://github.com/terrapower/armi/blob/0f32729b7504da8e8e676093eed2bedfa012ace1/armi/settings/fwSettings/globalSettings.py#L120-L126

So, this first `TODO` is just duplicate and less specific.

File: armi/bookkeeping/db/layout.py (hunk 1 above)
Comment (john-science) [reply armi/bookkeeping/db/layout.py:860]: I mean, imagine you need to "replace Nonsense with Nones", WHY would you want to use a function called "replace Nones with Nonsense"?  

That would be EXTRA confusing to explain to people.

Also... if that one function did both things I would want to break it up into two smaller functions that only do one thing each.

**Review by opotowsky — APPROVED** [review]
I stepped through each commit and they all make sense to me.

File: armi/settings/fwSettings/globalSettings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (john-science) [reply armi/settings/fwSettings/globalSettings.py:22]: Also, I have another PR open where I deal with these:  https://github.com/terrapower/armi/pull/1393

**Pull request merged**